### PR TITLE
v24.3.0-alpha.1 release notes

### PR DESCRIPTION
**Pull request by kathancox**
Fixes REL-1511

### CODE REVIEW

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md [hunk 1]
```diff
@@ -1,3 +1,511 @@
...
+
+<h3 id="v24-3-0-alpha-1-operational-changes">Operational changes</h3>
+
+- [Events]({% link v24.3/eventlog.md %}) `DiskSlownessDetected` and `DiskSlownessCleared` are now logged when disk slowness is detected and cleared on a store. [#127025][#127025]
+- Several [cluster settings]({% link v24.3/cluster-settings.md %}) allow you to configure rate-limiting traffic to cloud storage over various protocols. These settings begin with `cloudstorage`. [#127207][#127207]
+- The new [cluster setting]({% link v24.3/cluster-settings.md %}) `kv.range.range_size_hard_cap` allows you to limit how large a [range]({% link v24.3/architecture/overview.md %}#architecture-range) can grow before being split. The default is 8 GiB, which is 16 times the default max range size. If you have changed the max range size, you may need to adjust this cluster setting or reduce the range size. [#129450][#129450]
+- The following `kvflowcontrol` [metrics]({% link v24.3/metrics.md %}) have been renamed. After a cluster is finalized on v24.3, old and new metrics will be populated. The previous metrics under `kvasdmission.flow_controller` will be removed.
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:94]: @kvoli PTAL

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md [hunk 2]
```diff
@@ -1,3 +1,511 @@
...
+      `kvadmission.flow_controller.(regular|elastic)_wait_duration`        | `kvflowcontrol.eval_wait.(regular|elastic).duration`
+
+      [#130167][#130167]
+- The new `ranges.decommissioning` [metric]({% link v24.3/metrics.md %}) shows the number of ranges with a replica on a [decommissioning]({% link v24.3/node-shutdown.md %}) node. [#130117][#130117]
+- New [cluster settings]({% link v24.3/cluster-settings.md %}) have been added which control the refresh behavior for the cached data in Databases pages of the [DB Console]():
+      - `obs.tablemetadatacache.data_valid_duration`: the duration for which the data in `system.table_metadata` is considered valid before a cache reset will occur. Default: 20 minutes.
+      - `obs.tablemetadatacache.automatic_updates.enabled`: whether to automatically update the cache according the validity interval. Default: `false`.
+
+      [#130198][#130198]
+- New gauge [metrics]({% link v24.3/metrics.md %}) `security.certificate.expiration.{cert-type}` and `security.certificate.ttl.{cert-type}` show the expiration and TTL for a certificate. [#130110][#130110]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:117]: @angles-n-daemons PTAL

Comment (angles-n-daemons) [reply src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:117]: lgtm, thank you

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md (hunk 1 above)
Comment (kvoli) [reply src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:94]: LGTM

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md [hunk 3]
```diff
@@ -1,3 +1,513 @@
...
+
+- [Events]({% link v24.3/eventlog.md %}) `DiskSlownessDetected` and `DiskSlownessCleared` are now logged when disk slowness is detected and cleared on a store. [#127025][#127025]
+- Several [cluster settings]({% link v24.3/cluster-settings.md %}) allow you to configure rate-limiting traffic to cloud storage over various protocols. These settings begin with `cloudstorage`. [#127207][#127207]
+- The new [cluster setting]({% link v24.3/cluster-settings.md %}) `kv.range.range_size_hard_cap` allows you to limit how large a [range]({% link v24.3/architecture/overview.md %}#architecture-range) can grow before [backpressure]({% link v24.3/common-errors.md %}#split-failed-while-applying-backpressure-are-rows-updated-in-a-tight-loop) is applied. This can help to mitigate against a situation where a range cannot be split, such as when a range is comprised of a single key due to an issue with the schema or workload pattern or a bug in client application code. The default is 8 GiB, which is 16 times the default max range size. If you have changed the max range size, you may need to adjust this cluster setting or reduce the range size. [#129450][#129450]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:94]: @arulajmani PTAL

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md [hunk 4]
```diff
@@ -1,3 +1,513 @@
...
+- Added a LDAP authentication method to complement password-based login for the DB Console if HBA configuration has an entry for LDAP for the user attempting login, along with other matching criteria (like the requests originating IP address) for authentication to the DB Console. [#130418][#130418]
+- Added timers around key parts of the [changefeed]({% link v24.3/change-data-capture-overview.md %}) pipeline to help debug feeds experiencing issues. The `changefeed.stage.<stage>.latency` metrics now emit latency histograms for each stage. The metric respects the [changefeed `scope` label]({% link v24.3/monitor-and-debug-changefeeds.md %}#using-metrics-labels) for debugging specific feeds. [#128794][#128794]
+- For [enterprise changefeeds]({% link v24.3/how-does-an-enterprise-changefeed-work.md %}), [events]({% link v24.3/eventlog.md %}) `changefeed_failed` and `create_changefeed` now include a `JobId` field. [#131396][#131396]
+- The new [metric]({% link v24.3/metrics.md %}) `seconds_until_enterprise_license_expiry` allows you to monitor the status of a cluster's Enterprise license. [#129052][#129052].
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:62]: @angles-n-daemons PTAL

Comment (angles-n-daemons) [reply src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:62]: Perfect, thanks!

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md [hunk 5]
```diff
@@ -1,3 +1,340 @@
...
+- Upgrading to 24.3 is blocked if no license is installed, or if a trial/free license is installed with telemetry disabled. [#130576][#130576]
+- Changed the license `cockroach` is distributed under to the new CockroachDB Software License. [#131661][#131661]
+- Attempting to install a second Enterprise trial license on the same cluster will now fail. [#131422][#131422]
+- Changed the license `cockroach` is distributed under to the new CockroachDB Software License (CSL). [#131690][#131690]
+- Changed the license `cockroach` is distributed under to the new CockroachDB Software License (CSL). [#131686][#131686]
+- Changed the license `cockroach` is distributed under to the new CockroachDB Software License (CSL). [#131688][#131688]
+- Changed the license `cockroach` is distributed under to the new CockroachDB Software License (CSL). [#131687][#131687]
+- Changed the license `cockroach` is distributed under to the new CockroachDB Software License (CSL). [#131717][#131717]
+- Changed the license `cockroach` is distributed under to the new CockroachDB Software License (CSL). [#131689][#131689]
```
Comment (kathancox) [inline src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:32]: @rmloveland For each of these, should we update which license it was changed from as per the PR description?

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md [hunk 6]
```diff
@@ -1,3 +1,340 @@
...
+- The following `kvflowcontrol` [metrics]({% link v24.3/metrics.md %}) have been renamed. After a cluster is finalized on v24.3, old and new metrics will be populated. The previous metrics under `kvasdmission.flow_controller` will be removed.
+
+      Old metric names                                                     | New metric names
+      ---------------------------------------------------------------------|-----------------
+      `kvadmission.flow_controller.(regular|elastic)_tokens_available`     | `kvflowcontrol.tokens.eval.(regular|elastic).available`
```
Comment (kathancox) [inline src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:107]: @mdlinville I can't get this to render.

**Comment by kathancox** [comment]
@mdlinville There seems to be one final link breaking, looks like some missing anchor `href` but I'm not quite sure where it is.

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md [hunk 7]
```diff
@@ -1,3 +1,340 @@
...
 
-Release Date: TBD
+Release Date: October 9, 2024
+
+
+{% include releases/new-release-downloads-docker-image.md release=include.release %}
+<h3 id="v24-3-0-alpha-1-security-updates">Security updates</h3>
+
+- URLs in the [`CREATE CHANGEFEED`]({% link v24.3/create-changefeed.md %}) and [`CREATE SCHEDULE FOR CHANGEFEED`]({% link v24.3/create-schedule-for-changefeed.md %}) SQL statements are now sanitized of any secrets before being written to unredacted [logs]({% link v24.3/logging.md %}). [#126970][#126970]
+- The LDAP [cluster settings]({% link v24.3/cluster-settings.md %}) `server.ldap_authentication.client.tls_certificate` and `server.ldap_authentication.client.tls_key` do not have callbacks installed to reload the settings value for LDAP authManager. This change fixes this by adding the necessary callbacks. [#131151][#131151]
```
Comment (florence-crl) [inline src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:10]:
```suggestion
- The LDAP [cluster settings]({% link v24.3/cluster-settings.md %}) `server.ldap_authentication.client.tls_certificate` and `server.ldap_authentication.client.tls_key` did not have callbacks installed to reload the settings value for LDAP authManager. This change fixed this by adding the necessary callbacks. [#131151][#131151]
```

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md (hunk 6 above)
Comment (florence-crl) [reply src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:107]: From my testing, you can either have
(1) a pipe character by using `&#124;` or  &vert; WITHOUT the backticks for code formatting
OR
(2) have the backticks for code formatting AND replace the pipe with another character, like a forward slash

``` 
      Old metric names                                                     | New metric names
      ---------------------------------------------------------------------|-----------------
      kvadmission.flow_controller.(regular&#124;elastic)_tokens_available     | kvflowcontrol.tokens.eval.(regular&vert;elastic).available
      `kvadmission.flow_controller.(regular/elastic)_tokens_deducted`      | `kvflowcontrol.tokens.eval.(regular/elastic).deducted`
```

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md [hunk 8]
```diff
@@ -1,3 +1,340 @@
...
+      LDAP                  | `auth_ldap_conn_latency`
+      Password              | `auth_password_conn_latency`
+      SCRAM SHA-256         | `auth_scram_conn_latency`
+
+      [#131578][#131578]
+
+- Verbose logging of slow [Pebble]({% link v24.3/architecture/storage-layer.md %}#pebble) reads can no longer be enabled via the shorthand flag `--vmodule=pebble_logger_and_tracer=2`, where `pebble_logger_and_tracer` contains the CockroachDB implementation of the logger needed by Pebble. Instead, you must list the Pebble files that contain the log statements. For example  `--vmodule=reader=2,table=2`. [#127066][#127066]
+- The lowest admission control priority for the storage layer has been renamed from `ttl-low-pri` to `bulk-low-pri`. [#129564][#129564]
+- New clusters will now have a zone config defined for the `timeseries` range, which specifies `gc.ttlseconds` and inherits all other attributes from the zone config of the `default` range. This zone config will also be added to a cluster that is [upgraded]({% link v24.3/upgrade-cockroach-version.md %}) to v24.3 if it does not already have a zone config defined.[#128032][#128032]
```
Comment (florence-crl) [inline src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:144]:
```suggestion
- New clusters will now have a [zone configuration](]({% link v24.3/show-zone-configuration.md %}) defined for the `timeseries` range, which specifies `gc.ttlseconds` and inherits all other attributes from the zone config of the `default` range. This zone config will also be added to a cluster that is [upgraded]({% link v24.3/upgrade-cockroach-version.md %}) to v24.3 if it does not already have a zone config defined.[#128032][#128032]
```

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md [hunk 9]
```diff
@@ -1,3 +1,340 @@
...
+- Added new database pages that are available from the side navigation **Databases** link. [#131594][#131594]
+- The [DB Console]({% link v24.3/ui-overview.md %}) will reflect any throttling behavior from the cluster due to an expired license or missing telemetry data. Enterprise licenses are not affected. [#131326][#131326]
+- Users can hover over the node/region cell in multi-region deployments to view a list of nodes the database or table is on. [#130704][#130704]
+- The [**Databases** pages]({% link v24.3/ui-databases-page.md %}) in the DB console have been updated to read cached metadata about database and table storage statistics. The cache update time is now displayed in the top right-hand corner of the database and tables list pages. Users may trigger a cache refresh with the **refresh** icon next to the last updated time. The cache will also update automatically when users visit a **Databases** page and the cache is >= 20m old. [#131463][#131463]
```
Comment (florence-crl) [inline src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:161]:
```suggestion
- The [**Databases** pages]({% link v24.3/ui-databases-page.md %}) in the DB console have been updated to read cached metadata about database and table storage statistics. The cache update time is now displayed in the top right-hand corner of the database and tables list pages. Users may trigger a cache refresh with the **refresh** icon next to the last updated time. The cache will also update automatically when users visit a **Databases** page and the cache is older than or equal to 20 minutes. [#131463][#131463]
```

**Review by florence-crl — APPROVED** [review]
lgtm pending suggestions.

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md (hunk 6 above)
Comment (mdlinville) [reply src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md]: Thanks for showing me that workaround! I could also have wrapped the pipes within `{% raw %}...{% endraw %}`. I thought I was safe within backticks. Actrually I may have been, it looks like I also did not properly close the first cell in the table for example. In this situation the best solution is to fully-qualify the old and new metric names.

**Comment by mdlinville** [comment]
> @mdlinville There seems to be one final link breaking, looks like some missing anchor `href` but I'm not quite sure where it is.

It seems like this is good now? Let me run a local linkcheck just to be sure.

File: src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md (hunk 5 above)
Comment (kathancox) [reply src/current/_includes/releases/v24.3/v24.3.0-alpha.1.md:32]: OK cool, thanks Rich. It looked like there were a number of existing licenses. Your idea makes sense I will do this!

**Comment by mdlinville** [comment]
> @mdlinville There seems to be one final link breaking, looks like some missing anchor `href` but I'm not quite sure where it is.

Current state of this PR has no broken links locally (or in the PR checks). I think we are good on this bit.

**Pull request merged**